stdlib/Dates/src/io.jl

```diff
@@ -161,6 +173,11 @@ for (tok, fn) in zip("uU", [monthabbr, monthname])
     end
 end
 
+function format(io, d::DatePart{'p'}, dt, locale)
+    ampm = hour(dt) < 12 ? "AM" : "PM" # fixme: locale-specific?
```

Comment:
As far as I can tell from some googling, there are only a handful of english-speaking countries that use AM/PM, so yeah, I'm not quite sure how critical it is to support locale here. For reference (https://www.timeanddate.com/time/am-and-pm.html)